src/declarations/stencil-public-compiler.ts

```diff
@@ -2192,30 +2192,268 @@ export type OutputTarget =
+   * precache manifest. For more information, see the definition of `ignore` in the `glob`
+   * [documentation](https://github.com/isaacs/node-glob#options).
+   *
+   * @default ['node_modules/**']
+   */
   globIgnores?: string | string[];
-  templatedUrls?: any;
-  maximumFileSizeToCacheInBytes?: number;
-  manifestTransforms?: any;
-  modifyUrlPrefix?: any;
-  dontCacheBustURLsMatching?: RegExp;
-  navigateFallback?: string;
-  navigateFallbackWhitelist?: RegExp[];
-  navigateFallbackBlacklist?: RegExp[];
-  cacheId?: string;
-  skipWaiting?: boolean;
-  clientsClaim?: boolean;
-  directoryIndex?: string;
-  runtimeCaching?: any[];
-  ignoreUrlParametersMatching?: any[];
-  handleFetch?: boolean;
+
+  /**
+   * Files matching any of these patterns will be included in the precache
+   * manifest. For more information, see the
+   * [`glob` primer](https://github.com/isaacs/node-glob#glob-primer).
+   *
+   * @default ['**.{js,css,html}']
+   */
+  globPatterns?: string[] | undefined;
+
+  /**
+   * If true, an error reading a directory when generating a precache manifest
+   * will cause the build to fail. If false, the problematic directory will be
+   * skipped. For more information, see the definition of `strict` in the `glob`
+   * [documentation](https://github.com/isaacs/node-glob#options).
+   *
+   * @default true
+   */
+  globStrict?: boolean | undefined;
+
+  /**
+   * Any search parameter names that match against one of the RegExp in this array
+   * will be removed before looking for a precache match. This is useful if your
+   * users might request URLs that contain, for example, URL parameters used to
+   * track the source of the traffic.
+   *
+   * @default [/^utm_/]
+   */
+  ignoreURLParametersMatching?: RegExp[] | undefined;
+
+  /**
+   * A list of JavaScript files that should be passed to [`importScripts()`](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts)
+   * inside the generated service worker file. This is  useful when you want to
+   * let Workbox create your top-level service worker file, but want to include
+   * some additional code, such as a push event listener.
+   */
+  importScripts?: string[] | undefined;
+
+  /**
+   * Whether the runtime code for the Workbox library should be included in the
+   * top-level service worker, or split into a separate file that needs to be deployed
+   * alongside the service worker. Keeping the runtime separate means that users will
+   * not have to re-download the Workbox code each time your top-level service worker changes.
+   *
+   * @default false
+   */
+  inlineWorkboxRuntime?: boolean | undefined;
+
+  /**
+   * One or more functions which will be applied sequentially against the
+   * generated manifest. If `modifyURLPrefix` or `dontCacheBustURLsMatching` are
+   * also specified, their corresponding transformations will be applied first.
+   */
+  manifestTransforms?: any[] | undefined;
```

Comment:
Similar to my comment above, it looks like we narrow this type from `any` to `any[]`, which we might want to consider a breaking change as well